adapters/syslog/syslog.go

```diff
@@ -81,104 +158,80 @@ func NewSyslogAdapter(route *router.Route) (router.LogAdapter, error) {
 	if err != nil {
 		return nil, err
 	}
+	connIsUDP := reflect.TypeOf(conn).String() == "*net.UDPConn"
```

Comment:
type assertion is preferred in this case for performance/complexity reasons. this can become:
```
	var connIsUDP bool
	switch conn.(type) {
	case *net.UDPConn:
		connIsUDP = true
	default:
		connIsUDP = false
	}
	debug("setting connIsUDP to:", connIsUDP)
```

also this should be it's own function.

Reply:
Yes if type assertion is preferred I agree it's own function is better.

Will restore that as suggested. Btw the funcion name is mispelled so I will correct that too.